js/initcontext.go

```diff
@@ -69,6 +72,9 @@ type InitContext struct {
 	compatibilityMode lib.CompatibilityMode
 
 	logger logrus.FieldLogger
+
+	shares     map[string]sharedArray
+	sharesLock *sync.Mutex
```

Comment:
Somewhat connected to https://github.com/loadimpact/k6/pull/1739#discussion_r540121267, maybe this should be more generic. Basically, a way VUs can share all sorts of objects between them in a `sync.Once` fashion, with each object being referenced by its string name/path. So, basically:
- move this to https://github.com/loadimpact/k6/blob/master/js/common/initenv.go#L33 and leave it private
- use `sync.Map` instead of this custom one
- and have generic `GetOrCreateShare(name string, createCallback func() interface{}) interface{}` method 

In the `SharedArray` constructor you can easily type assert that the value you get is a sharedArray, but extensions would be easily able to use it for other purposes as well, e.g. shared atomic counters or whatever.

Reply:
earlier it was map[string]interface{} so ... I can use it for the sharedObject ... but I decided to drop it and leave it to be extended when we actually need to ... So I am again for just leaving it until we actually need it, and at that point whoever does the change can better gauge if that is still better

Reply:
> until we actually need it

I gave you an example of how it can be useful - extensions would be able to share state between al VUs, for example as a more generic version of https://github.com/MStoykov/xk6-counter

And given that it doesn't really cost you anything to switch it, I don't see a reason not to. And sync.Map probably is much better for this use case anyway:
>  The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex or RWMutex.

Reply:
see e94190c as the comment says ... using sync.Map isn't really useful :(

Reply:
:+1: yeah, makes sense when it can't accept a callback... :disappointed: and we can't even use a `RWMutex` because of that behavior... oh well :man_shrugging: